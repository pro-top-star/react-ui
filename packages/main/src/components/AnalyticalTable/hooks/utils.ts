export const makeTemplateColumns = (columns, resizedColumns) => {
  return columns
    .map((column) => {
      if (resizedColumns[column.id]) {
        return `${resizedColumns[column.id]}px`;
      }
      return `minmax(${column.minWidth}px, ${column.width || '1fr'})`;
    })
    .join(' ');
};
