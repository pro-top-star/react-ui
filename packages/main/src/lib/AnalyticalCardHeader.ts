import { AnalyticalCardHeader } from '../components/AnalyticalCard/header/AnalyticalCardHeader';

export { AnalyticalCardHeader };
