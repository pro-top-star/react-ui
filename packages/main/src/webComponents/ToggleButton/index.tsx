import { Event } from '@ui5/webcomponents-react-base/lib/Event';
import { ButtonDesign } from '@ui5/webcomponents-react/lib/ButtonDesign';
import { withWebComponent } from '@ui5/webcomponents-react/lib/withWebComponent';
import UI5ToggleButton from '@ui5/webcomponents/dist/ToggleButton';
import React, { FC } from 'react';
import { WithWebComponentPropTypes } from '../../internal/withWebComponent';

export interface ToggleButtonPropTypes extends WithWebComponentPropTypes {
  design?: ButtonDesign; // @generated
  disabled?: boolean; // @generated
  icon?: string; // @generated
  iconEnd?: boolean; // @generated
  submits?: boolean; // @generated
  pressed?: boolean; // @generated
  onClick?: (event: Event) => void; // @generated
  children?: string; // @generated
}

/**
 * <code>import { ToggleButton } from '@ui5/webcomponents-react/lib/ToggleButton';</code>
 * <br />
 * <a href="https://sap.github.io/ui5-webcomponents/playground/components/ToggleButton" target="_blank">UI5 Web Components Playground</a>
 */
const ToggleButton: FC<ToggleButtonPropTypes> = withWebComponent<ToggleButtonPropTypes>(UI5ToggleButton);

ToggleButton.displayName = 'ToggleButton';

ToggleButton.defaultProps = {
  design: ButtonDesign.Default // @generated
};

export { ToggleButton };
