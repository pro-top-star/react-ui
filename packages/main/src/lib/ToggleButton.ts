import { ToggleButton } from '../webComponents/ToggleButton';

export { ToggleButton };
