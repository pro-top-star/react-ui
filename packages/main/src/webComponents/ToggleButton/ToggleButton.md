[Link to UI5 Web Components Playground](https://sap.github.io/ui5-webcomponents/playground/components/ToggleButton)

```jsx
import { ToggleButton } from '@ui5/webcomponents-react/lib/ToggleButton';
```
