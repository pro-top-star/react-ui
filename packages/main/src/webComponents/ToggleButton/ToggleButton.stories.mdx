import { ArgsTable, Canvas, Meta, Story } from '@storybook/addon-docs/blocks';
import { ToggleButton } from '@ui5/webcomponents-react/dist/ToggleButton';
import { ButtonDesign } from '@ui5/webcomponents-react/dist/ButtonDesign';
import '@ui5/webcomponents-icons/dist/employee.js';
import { createSelectArgTypes } from '@shared/stories/createSelectArgTypes';
import { DocsHeader } from '@shared/stories/DocsHeader';
import { DocsCommonProps } from '@shared/stories/DocsCommonProps';

<Meta
  title="Inputs / ToggleButton"
  component={ToggleButton}
  argTypes={{
    ...createSelectArgTypes({ design: ButtonDesign }),
    children: { control: 'text' },
    ...DocsCommonProps
  }}
  args={{
    design: ButtonDesign.Default,
    icon: 'employee',
    children: 'ToggleButton Text',
    style: {},
    className: '',
    tooltip: '',
    slot: '',
    ref: null
  }}
/>

<DocsHeader />

<br />

## Example

<Canvas>
  <Story name="Default">
    {(args) => {
      return <ToggleButton {...args} />;
    }}
  </Story>
</Canvas>

## Properties

<ArgsTable story="." />
