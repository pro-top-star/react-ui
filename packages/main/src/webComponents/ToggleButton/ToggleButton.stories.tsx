import { action } from '@storybook/addon-actions';
import { boolean, select, text } from '@storybook/addon-knobs';
import { ButtonDesign } from '@ui5/webcomponents-react/lib/ButtonDesign';
import { ToggleButton } from '@ui5/webcomponents-react/lib/ToggleButton';
import React from 'react';

export default {
  title: 'UI5 Web Components / ToggleButton',
  component: ToggleButton
};

export const generatedDefaultStory = () => (
  <ToggleButton
    pressed={boolean('pressed', false)}
    design={select('design', ButtonDesign, ButtonDesign.Default)}
    disabled={boolean('disabled', false)}
    icon={text('icon', '')}
    iconEnd={boolean('iconEnd', false)}
    submits={boolean('submits', false)}
    onClick={action('onClick')}
  >
    Some Content
  </ToggleButton>
);

generatedDefaultStory.story = {
  name: 'Generated default story'
};
