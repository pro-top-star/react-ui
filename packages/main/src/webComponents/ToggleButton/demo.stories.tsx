import { action } from '@storybook/addon-actions';
import { boolean, select } from '@storybook/addon-knobs';
import { ButtonDesign } from '@ui5/webcomponents-react/lib/ButtonDesign';
import { ToggleButton } from '@ui5/webcomponents-react/lib/ToggleButton';
import React from 'react';
import notes from './ToggleButton.md';

export default {
  title: 'UI5 Web Components / ToggleButton',
  component: ToggleButton,
  parameters: { notes }
};

export const generatedDefaultStory = () => (
  <ToggleButton
    design={select('design', ButtonDesign, null)}
    disabled={boolean('disabled', false)}
    icon="add"
    iconEnd={boolean('iconEnd', false)}
    pressed={boolean('pressed', false)}
    onClick={action('onClick')}
  >
    Some Content
  </ToggleButton>
);

generatedDefaultStory.story = {
  name: 'Generated default story'
};
