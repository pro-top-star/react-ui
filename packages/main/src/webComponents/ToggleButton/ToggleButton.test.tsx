import { mountThemedComponent } from '@shared/tests/utils';
import { ToggleButton } from '@ui5/webcomponents-react/lib/ToggleButton';
import React from 'react';

describe('ToggleButton', () => {
  test('Basic Test (generated)', () => {
    const wrapper = mountThemedComponent(<ToggleButton />);
    expect(wrapper.render()).toMatchSnapshot();
  });
});
