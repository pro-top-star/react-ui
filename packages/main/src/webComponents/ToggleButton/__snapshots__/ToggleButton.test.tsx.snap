// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleButton Basic Test (generated) 1`] = `
<DocumentFragment>
  <ui5-togglebutton
    design="Default"
    ui5-togglebutton=""
  />
</DocumentFragment>
`;
